mos-platform/nes-mmc1/bank.h

```diff
@@ -97,21 +116,65 @@ void set_chr_bank_1_retry(char bank_id);
 // split(0); ---- wait for sprite zero hit, set X scroll to 0
 // split_chr_bank_0(6) ---- change CHR bank to #6
 
-#define MIRROR_LOWER_BANK 0
-#define MIRROR_UPPER_BANK 1
-#define MIRROR_VERTICAL 2
-#define MIRROR_HORIZONTAL 3
+enum Mirroring {
+  MIRROR_LOWER_BANK = 0,
+  MIRROR_UPPER_BANK,
+  MIRROR_VERTICAL,
+  MIRROR_HORIZONTAL,
+};
 
 // Set the current mirroring mode. Your options are MIRROR_LOWER_BANK,
 // MIRROR_UPPER_BANK, MIRROR_HORIZONTAL, and MIRROR_VERTICAL.
-// LOWER and UPPER are single screen modes.
-__attribute__((leaf)) void set_mirroring(char mirroring);
-
-// Set all 5 bits of the $8000 MMC1 Control register. Overwrites mirroring
-// setting.
+// LOWER and UPPER are single screen modes. Applies immediately and reapplies at
+// each NMI. May have no immediate effect if interrupted by NMI.
+void set_mirroring(enum Mirroring mirroring);
+
+// Set the mirroring temporarily, but override with the previous setting at next
+// NMI. May have no effect if interrupted by NMI.
+void split_mirroring(enum Mirroring mirroring);
+
+// Set the mirroring to be applied automatically at the next NMI.
+void defer_mirroring(enum Mirroring mirroring);
+
+enum ChrRomBankMode {
+  CHR_ROM_BANK_MODE_8 = 0b00000,
+  CHR_ROM_BANK_MODE_4 = 0b10000,
+};
+
+// Set the current CHR-ROM bank mode. Applies immediately and reapplies at
+// each NMI. May have no immediate effect if interrupted by NMI.
+void set_chr_rom_bank_mode(enum ChrRomBankMode mode);
```

Comment:
Why not `set_chr_bank_mode`? CHR can be both ROM or RAM here.

Reply:
Ah, excellent point.